server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,436 @@
+        clusterSettings.addSettingsUpdateConsumer(
+            CLUSTER_ROUTING_ALLOCATION_AWARENESS_FORCE_GROUP_SETTING,
+            this::setForcedAwarenessAttributes
+        );
+    }
+
+    private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        this.awarenessAttributes = awarenessAttributes;
+    }
+
+    private void setForcedAwarenessAttributes(Settings forceSettings) {
+        Map<String, List<String>> forcedAwarenessAttributes = new HashMap<>();
+        Map<String, Settings> forceGroups = forceSettings.getAsGroups();
+        for (Map.Entry<String, Settings> entry : forceGroups.entrySet()) {
+            List<String> aValues = entry.getValue().getAsList("values");
+            if (aValues.size() > 0) {
+                forcedAwarenessAttributes.put(entry.getKey(), aValues);
+            }
+        }
+        this.forcedAwarenessAttributes = forcedAwarenessAttributes;
+    }
+
+    public void initiateAttributeDecommissioning(
+        final DecommissionAttribute decommissionAttribute,
+        final ActionListener<ClusterStateUpdateResponse> listener,
+        ClusterState state
+    ) {
+        // validates if the correct awareness attributes and forced awareness attribute set to the cluster before initiating decommission
+        // action
+        validateAwarenessAttribute(decommissionAttribute, awarenessAttributes, forcedAwarenessAttributes);
+
+        Set<DiscoveryNode> clusterManagerNodesToBeDecommissioned = nodesWithDecommissionAttribute(state, decommissionAttribute, true);
+        ensureNoQuorumLossDueToDecommissioning(
+            decommissionAttribute,
+            clusterManagerNodesToBeDecommissioned,
+            state.getLastAcceptedConfiguration()
+        );
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().custom(DecommissionAttributeMetadata.TYPE);
+        // validates that there's no inflight decommissioning or already executed decommission in place
+        ensureNoAwarenessAttributeDecommissioned(decommissionAttributeMetadata, decommissionAttribute);
+
+        logger.info("initiating awareness attribute [{}] decommissioning", decommissionAttribute.toString());
+
+        // remove all 'to-be-decommissioned' cluster manager eligible nodes from voting config
+        // The method ensures that we don't exclude same nodes multiple times
+        excludeDecommissionedClusterManagerNodesFromVotingConfig(clusterManagerNodesToBeDecommissioned);
+
+        // explicitly throwing NotClusterManagerException as we can certainly say the local cluster manager node will
+        // be abdicated and soon will no longer be cluster manager.
+        if (transportService.getLocalNode().isClusterManagerNode()
+            && !nodeHasDecommissionedAttribute(transportService.getLocalNode(), decommissionAttribute)) {
+            registerDecommissionAttribute(decommissionAttribute, listener);
+        } else {
+            throw new NotClusterManagerException(
```

Comment:
We need this particular exception in place so that the transport call is retried again. Today, any transport call is retried again only when it gets `FailedToCommitClusterStateException` or `NotClusterManagerException`. Either we go ahead and create a new exception for this case and update the retry logic to handle this new exception for retry in `TransportClusterManagerNodeAction`. Or, we use `NotClusterManagerException`, as we can certainly that the active leader will be for sure abdicated.

Went ahead with approach 2 as didn't want to make any core change for this service. I can add a comment to explain why we are explicitly throwing that.

Reply:
Update - we don't need to throw the exception explicitly. If we are certain that the local node is a decommissioned leader, we can call the listener with `NotClusterManagerException`